manual/src/refman/extensions/arrayliterals.etex

```diff
@@ -0,0 +1,74 @@
+(Introduced in OCaml 5.4)
+
+Since OCaml 5.4, array literal syntax @"[|" e1 ';' e2 ';' ... ';' eN "|]"@ can
+be used to denote values of type "'a array" as well as "floatarray", both in
+expression and pattern positions. This syntax is also used to display "floatarray" values in the toplevel.
+
+The compiler will use contextual information to infer the type of the array
+literal. However, in the absence of any specific contextual information, it will
+be assumed that the array literal is of type "'a array":
+
+\begin{caml_example}{verbatim}
+let x = [|42.|];;
+\end{caml_example}
+
+The compiler will exploit the knowledge it gleans from the context of the array
+literal to decide which type to assign to it. In the following examples,
+the array literal is assigned type "floatarray":
```

Comment:
I have the impression that this sentence is redundant once we state clearly that the compiler is using the expected type. I would keep the examples but move them before the "defaulting to 'a array".

Reply:
I removed the sentence (as suggested by @goldfirere as well).